### PR TITLE
Wayfinder init local docker tests

### PR DESCRIPTION
**Pull request by dallascrichmond**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

### CODE REVIEW

File: src/api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM node:18.16.0
+
+# Create api directory
+WORKDIR /api
+
+# Install app dependencies
+# A wildcard is used to ensure both package.json AND package-lock.json are copied
+# where available (npm@5+)
+COPY package*.json ./
```
Comment (BradyMitch) [inline src/api/Dockerfile:9]: package-lock.json is auto-generated based on your package.json so it shouldnt be copied into the container. You can add it to the dockerignore file and only `COPY . .` (one less copy step, no need to copy package separately).

Then once you start with multi-stage and targeted builds I can show you how using `ENV NODE_ENV=production` will tell the `npm install` command to omit dev dependencies.

Comment (BradyMitch) [reply src/api/Dockerfile:9]: You would also want to add package-lock.json to the gitignore. You may also have to `git rm package-lock.json` so github stops keeping track of it.

It's also important that you don't introduce any more of those caret symbols (^) into the package.json and only specify exact version numbers for packages. Make sure everyone on your team has run the command `npm config set save-exact true`.

Comment (dallascrichmond) [reply src/api/Dockerfile]: All done and pushed.

File: src/api/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+node_modules
+npm-debug.log
+package-lock.json
```
Comment (BradyMitch) [inline src/api/.dockerignore:3]:
```suggestion
package-lock.json

```

File: src/api/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+FROM node:18.16.0
+
+# Create api directory
+WORKDIR /api
+
+# Install app dependencies
+# A wildcard is used to ensure both package.json AND package-lock.json are copied
```
Comment (BradyMitch) [inline src/api/Dockerfile:7]:
```suggestion
```
Do your best to keep your comments and documentation up to date

File: src/api/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+# where available (npm@5+)
+COPY . .
+
+RUN npm install
+# If you are building your code for production
+# RUN npm ci --omit=dev
+
+# Bundle app source
+COPY . .
```
Comment (BradyMitch) [inline src/api/Dockerfile:16]:
```suggestion
```
This one can be removed now since we're copying everything on an earlier line.

**Comment by akroon3r** [comment]
Make sure to click "resolve conversation" when you've addressed feedback and it's been confirmed.

It also looks like we've got some merge conflicts, so lets get those addressed before we merge to main. If you need help with doing so, message me and we can do it over a call.

File: src/api/.dockerignore (hunk 2 above)
Comment (dallascrichmond) [reply src/api/.dockerignore]: Change made

File: src/api/Dockerfile (hunk 3 above)
Comment (dallascrichmond) [reply src/api/Dockerfile]: Took out wildcard comment

File: src/api/Dockerfile (hunk 4 above)
Comment (dallascrichmond) [reply src/api/Dockerfile]: Removed the extra COPY . .

File: src/api/.gitignore
Comment (Bottle7) [inline src/api/.gitignore:1]: .gitignore only needs to exist at the project root. Use `**/node_modules` syntax to ignore within sub-directories as well.

File: src/api/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+# Create api directory
+WORKDIR /api
+
+# Install app dependencies
+# where available (npm@5+)
+COPY . .
+
+RUN npm install
+# If you are building your code for production
```
Comment (Bottle7) [inline src/api/Dockerfile:11]: You don't need these comments. Typically, a separate Dockerfile will be made for production or dev.

File: src/api/.gitignore
Comment (dallascrichmond) [reply src/api/.gitignore:1]: Syntax has been changed and .gitignore has been moved to the project root

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+.env
+**/node_modules
+dist
+package-lock.json
```
Comment (Bottle7) [inline .gitignore:4]: You're goning to want to add the wildcards to the package-lock and dist just like the node_modules.

**Review by Bottle7 — APPROVED** [review]

**Pull request merged**